lib/sockfd_cache.c

```diff
@@ -517,3 +637,45 @@ void sockfd_cache_del(const struct node_id *nid, struct sockfd *sfd)
 	sockfd_cache_del_node(nid);
 	free(sfd);
 }
+
+static void *monitor_sd_node_connectivity(void *ignored)
+{
+	int err;
+
+	sd_info("node connectivity monitor main loop");
+
+	for (;;) {
+		struct sockfd_cache_entry *entry;
+		rb_for_each_entry(entry, &sockfd_cache.root, rb) {
```

Comment:
@mitake Sure, if you like, you can merge this PR now and make the changes after. I will also look into it once I get time. I think the locking mechanism might be tricky. Sheep may hang if we put a wrong lock. Ie. Cascaded write or Read/Write mix locks.